### PR TITLE
Only check for scene_exception when series_name is known.

### LINKED ISSUE #4941: [APP SUBMITTED]: AttributeError: 'NoneType' object has no attribute 'lower'
### INFO
**Python Version**: `2.7.14 (default, Dec 14 2017, 15:51:29) [GCC 6.4.0]`
**Operating System**: `Linux-4.9.0-8-amd64-x86_64-with`
**Locale**: `None`
**Branch**: [master](../tree/master)
**Database**: `44.12`
**Commit**: pymedusa/Medusa@f5973b28465246e67465d7d86bdc8aa74d13e659
**Link to Log**: https://gist.github.com/fc5b49522882a34aa5c6c23b8ea16c36
### ERROR
<pre>
2018-08-21 22:50:35 ERROR    SHOWQUEUE-REFRESH :: [f5973b2] 71863: Error while refreshing show Lupin III. Error: 'NoneType' object has no attribute 'lower'
Traceback (most recent call last):
  File "/app/medusa/<a href="../blob/f5973b28465246e67465d7d86bdc8aa74d13e659/medusa/show_queue.py#L686">medusa/show_queue.py</a>", line 686, in run
    self.show.refresh_dir()
  File "/app/medusa/<a href="../blob/f5973b28465246e67465d7d86bdc8aa74d13e659/medusa/tv/series.py#L1718">medusa/tv/series.py</a>", line 1718, in refresh_dir
    self.load_episodes_from_dir()
  File "/app/medusa/<a href="../blob/f5973b28465246e67465d7d86bdc8aa74d13e659/medusa/tv/series.py#L1006">medusa/tv/series.py</a>", line 1006, in load_episodes_from_dir
    parse_result = NameParser(series=self, try_indexers=True).parse(ep_file_name)
  File "/app/medusa/<a href="../blob/f5973b28465246e67465d7d86bdc8aa74d13e659/medusa/name_parser/parser.py#L366">medusa/name_parser/parser.py</a>", line 366, in parse
    result = self._parse_string(name)
  File "/app/medusa/<a href="../blob/f5973b28465246e67465d7d86bdc8aa74d13e659/medusa/name_parser/parser.py#L297">medusa/name_parser/parser.py</a>", line 297, in _parse_string
    new_episode_numbers, new_season_numbers, new_absolute_numbers = self._parse_anime(result)
  File "/app/medusa/<a href="../blob/f5973b28465246e67465d7d86bdc8aa74d13e659/medusa/name_parser/parser.py#L189">medusa/name_parser/parser.py</a>", line 189, in _parse_anime
    scene_season = scene_exceptions.get_scene_exceptions_by_name(result.series_name)[0][1]
  File "/app/medusa/<a href="../blob/f5973b28465246e67465d7d86bdc8aa74d13e659/medusa/scene_exceptions.py#L173">medusa/scene_exceptions.py</a>", line 173, in get_scene_exceptions_by_name
    if show_name.lower() in show_names:
AttributeError: 'NoneType' object has no attribute 'lower'
</pre>
---
_STAFF NOTIFIED_: @pymedusa/support @pymedusa/moderators

### PR DESCRIPTION
**Pull request by p0psicles**
When parsing an episode like Lupin III/Season 02/S2E22.mkv, it will parse it in two goes.
First attempt parsing `Lupin III/Season 02/S2E22.mkv` and then `S2E22.mkv`. The second parse, cannot resolve it to a series name.
And therefor the scene_exceptions call resulted in an exception.

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

Should fix #4941
@medariox can you double check the solution?

Alternative could be changing to
`scene_season = scene_exceptions.get_scene_exceptions_by_name(result.series_name or result.series.name)[0][1]`
in parser.py:L189

### CODE REVIEW

**Comment by medariox** [comment]
I would say the one thing doesn't exclude the other. The fix is surely needed, but at this point I'm not sure what's the difference between `result.series.name` and `result.series_name`. Shouldn't both be equal?

**Comment by p0psicles** [comment]
I think that result.series is available if the parsed title can be mapped to a show in the library.

Result.series_name is the raw show title parsed by guessit.

**Comment by medariox** [comment]
So my assumption should correct, if `result.series_name` is None, `result.series.name` must be non as well. That's why I don't think the or is necessary. About the fix: if None is passed there is still a lot going on. I think that maybe an early exit would be more appropriate in this case.

**Comment by p0psicles** [comment]
In this case series_name was none, but series.name was available and matched to the show lupin 3

**Comment by medariox** [comment]
Hmm, will have a closer look at that code tomorrow.

File: medusa/scene_exceptions.py [hunk 1]
```diff
@@ -140,6 +140,12 @@ def get_scene_exceptions_by_name(show_name):
     # TODO: Rewrite to use exceptions_cache since there is no need to hit db.
     # TODO: Make the query more linient. For example. `Jojo's Bizarre Adventure Stardust Crusaders` will not match
     # while `Jojo's Bizarre Adventure - Stardust Crusaders` is available.
+    if show_name is None:
+        logger.debug(
+            'Scene exception lookup failed because no show name was provided'
+        )
```
Comment (sharkykh) [inline medusa/scene_exceptions.py]: @medariox Maybe make this a one-liner?

Comment (medariox) [reply medusa/scene_exceptions.py]: I'm okay with the change, but you'll need to push the change or someone else. Not at a PC until tomorrow.

**Pull request merged**